src/main/java/dinkplugin/SettingsManager.java

```diff
@@ -0,0 +1,127 @@
+package dinkplugin;
+
+import dinkplugin.notifiers.CollectionNotifier;
+import dinkplugin.notifiers.CombatTaskNotifier;
+import dinkplugin.util.Utils;
+import lombok.RequiredArgsConstructor;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.Varbits;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.events.ConfigChanged;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.VisibleForTesting;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+@Slf4j
+@Singleton
+@RequiredArgsConstructor(onConstructor_ = { @Inject })
+public class SettingsManager {
+    private static final String CONFIG_GROUP = "dinkplugin";
+    private static final Pattern DELIM = Pattern.compile("[,;\\n]");
+
+    private final Collection<String> nameDenyList = new HashSet<>();
+
+    private final Client client;
+    private final ClientThread clientThread;
+    private final DinkPlugin plugin;
+    private final DinkPluginConfig config;
+
+    @Synchronized
+    public boolean testUsername(String name) {
```

Comment:
I'm not sold on this function name, especially not without a comment

Reply:
Changed to `isNamePermitted` (with javadoc) in https://github.com/pajlads/DinkPlugin/pull/114/commits/922c8661550ad5c9b77de35b98e254214443dc4d